python_modules/libraries/dagster-dbt/dagster_dbt/source_freshness_builder.py

```diff
@@ -0,0 +1,279 @@
+        if error_lower_bound_delta
+        else None,
+    }
+
+    @freshness_multi_asset_check(
+        params_metadata=JsonMetadataValue(params_metadata), asset_keys=asset_keys, blocking=blocking
+    )
+    def the_check(context: AssetCheckExecutionContext) -> Iterable[AssetCheckResult]:
+        for check_key in context.selected_asset_check_keys:
+            asset_key = check_key.asset_key
+            current_timestamp = get_current_timestamp()
+            current_time_in_freshness_tz = datetime.datetime.fromtimestamp(
+                current_timestamp, tz=get_timezone(timezone)
+            )
+
+            # Calculate both lower bounds
+            warn_time_lower_bound = None
+            if warn_lower_bound_delta:
+                warn_time_lower_bound = current_time_in_freshness_tz - warn_lower_bound_delta
+
+            error_time_lower_bound = None
+            if error_lower_bound_delta:
+                error_time_lower_bound = current_time_in_freshness_tz - error_lower_bound_delta
+
+            # Retrieve the last update record
+            latest_record = retrieve_last_update_record(
+                instance=context.instance, asset_key=asset_key, partition_key=None
+            )
+            update_timestamp = (
+                retrieve_timestamp_from_record(latest_record) if latest_record else None
+            )
+
+            # Determine if check passes and what severity to use
+            passed = True
+            severity: AssetCheckSeverity = AssetCheckSeverity.ERROR
+
+            # Determine the most restrictive lower bound for metadata
+            metadata_lower_bound = None
+            if update_timestamp is not None:
+                if error_lower_bound_delta and error_time_lower_bound:
+                    if update_timestamp < error_time_lower_bound.timestamp():
+                        passed = False
+                        severity = AssetCheckSeverity.ERROR
+                        metadata_lower_bound = error_time_lower_bound
+                    elif (
+                        warn_lower_bound_delta
+                        and warn_time_lower_bound
+                        and update_timestamp < warn_time_lower_bound.timestamp()
+                    ):
+                        passed = False
+                        severity = AssetCheckSeverity.WARN
+                        metadata_lower_bound = warn_time_lower_bound
+                    else:
+                        metadata_lower_bound = (
+                            warn_time_lower_bound
+                            if warn_time_lower_bound
+                            else error_time_lower_bound
+                        )
+                elif warn_lower_bound_delta and warn_time_lower_bound:
+                    if update_timestamp < warn_time_lower_bound.timestamp():
+                        passed = False
+                        severity = AssetCheckSeverity.WARN
+                    metadata_lower_bound = warn_time_lower_bound
```

Comment:
I believe it could be simplified as this

```suggestion
                if (
                    error_lower_bound_delta 
                    and error_time_lower_bound 
                    and update_timestamp < error_time_lower_bound.timestamp()
                ):
                    passed = False
                    severity = AssetCheckSeverity.ERROR
                    metadata_lower_bound = error_time_lower_bound
               elif (
                    warn_lower_bound_delta
                    and warn_time_lower_bound
                    and update_timestamp < warn_time_lower_bound.timestamp()
                ):
                    passed = False
                    severity = AssetCheckSeverity.WARN
                    metadata_lower_bound = warn_time_lower_bound
                else:
                    metadata_lower_bound = (
                        warn_time_lower_bound
                        if warn_time_lower_bound
                        else error_time_lower_bound
                    )
```